controllers/argocd/util.go

```diff
@@ -974,8 +981,31 @@ func setResourceWatches(bldr *builder.Builder, clusterResourceMapper, tlsSecretM
 		},
 	}
 
+	// Add new predicate to delete Notifications Resources. The predicate watches the Argo CD CR for changes to the `.spec.Notifications.Enabled`
+	// field. When a change is detected that results in notifications being disabled, we trigger deletion of notifications resources
+	deleteNotificationsPred := predicate.Funcs{
+		UpdateFunc: func(e event.UpdateEvent) bool {
+			newCR, ok := e.ObjectNew.(*argoprojv1a1.ArgoCD)
+			if !ok {
+				return false
+			}
+			oldCR, ok := e.ObjectOld.(*argoprojv1a1.ArgoCD)
+			if !ok {
+				return false
+			}
+			if (oldCR.Spec.Notifications.Enabled != newCR.Spec.Notifications.Enabled) && !newCR.Spec.Notifications.Enabled {
```

Comment:
I guess this condition can be simplified to `oldCR.Spec.Notifications.Enabled && !newCR.Spec.Notifications.Enabled`. WDYT?

Reply:
will do :+1: